### ISSUE TITLE
opentofu-1.6.0-beta3 fails on refreshing existing resources (AWS)

### ISSUE THREAD

**Issue by sdlarsen** [opentofu/opentofu#988]
### OpenTofu Version

```shell
OpenTofu v1.6.0-beta3
on linux_amd64
+ provider registry.opentofu.org/community-terraform-providers/ignition v2.2.2
+ provider registry.opentofu.org/datadog/datadog v3.33.0
+ provider registry.opentofu.org/gitlabhq/gitlab v16.6.0
+ provider registry.opentofu.org/hashicorp/archive v2.4.0
+ provider registry.opentofu.org/hashicorp/aws v5.24.0
+ provider registry.opentofu.org/hashicorp/local v2.4.0
+ provider registry.opentofu.org/hashicorp/null v3.2.1
+ provider registry.opentofu.org/hashicorp/template v2.2.0
+ provider registry.opentofu.org/hashicorp/tls v4.0.4
+ provider registry.opentofu.org/opentofu/aws v5.30.0
+ provider registry.opentofu.org/opentofu/null v3.2.2
```

### OpenTofu Configuration Files

```hcl
...tofu config...
```

### Debug Output

doing tofu plan shows:

```
╷
│ Error: reading Route53 Hosted Zone (redacted): AccessDenied: User: arn:aws:iam::redacted:user/redacted is not authorized to access this resource
│ 	status code: 403, request id: 2644fe40-4a87-438c-ba1c-f22a1cf990c9
│
│   with module.cluster.module.dns.aws_route53_zone.private,
│   on ../../../providers/aws/dns/main.tf line 2, in resource "aws_route53_zone" "private":
│    2: resource "aws_route53_zone" "private" {
│
╵
╷
│ Error: reading Amazon S3 (Simple Storage) Bucket (redacted): operation error S3: HeadBucket, https response error StatusCode: 403, RequestID: 3PF2Z7R354Q5KJJC, HostID: nb8QG0bj+68s5X10hSsBun+8bq7yV75KQTQu4NPzzzuc3lo+C2FMouvV34NVoKCx/hsKN8WOnYk=, api error Forbidden: Forbidden
│
│   with module.cluster.module.ignition-s3.aws_s3_bucket.ignition,
│   on ../../../providers/aws/s3/ignition/main.tf line 2, in resource "aws_s3_bucket" "ignition":
│    2: resource "aws_s3_bucket" "ignition" {
│
╵
```

### Expected Behavior

On tofu 1.6.0-beta2 no errors are shown (and this is what was used to create the resources)

### Actual Behavior

Fails with the above

### Steps to Reproduce

tofu plan

### Additional Context

No changes to my config. Downgrading to beta2 and all is well.

**Comment by cam72cam** [comment]
That's fascinating. On initial inspection I'm not seeing any changes that seem like they would be directly related: https://github.com/opentofu/opentofu/compare/v1.6.0-beta2...v1.6.0-beta3

However I am seeing:
`+ provider registry.opentofu.org/hashicorp/aws v5.24.0` and
`+ provider registry.opentofu.org/opentofu/aws v5.30.0` 

That looks like a good place to start, and could be related to changes in how we handle the default namespace.

Could you check on beta2 to see what providers and versions are listed?

**Comment by cam72cam** [comment]
## TL;DR:  A versioned "hashicorp/aws" provider will no longer apply version constraints to "aws" from a submodule.  We may need to add some deeper equivalence to "hashicorp" and "opentofu" provider namespaces in tofu to fix this regression.

Reproduced with:
main.tf using a non-versioned hashicorp/tfcoremock
```hcl
terraform {
        required_providers {
                foo = {
                        source = "hashicorp/tfcoremock"
                }
        }
}


module "mod" {
        source = "./sub"
}
```

sub/main.tf using a versioned hashicorp/tfcoremock
```hcl
variable "test" {
        type = string
}

terraform {
        required_providers {
                tfcoremock = {
                        source = "hashicorp/tfcoremock"
                        version = "0.1.3"
                }
        }
}

resource "tfcoremock_simple_resource" "resource" {
    provider = tfcoremock.secondary
}
```

```bash
~/go/bin/tofu -version
OpenTofu v1.6.0-dev
on linux_amd64
+ provider registry.opentofu.org/hashicorp/tfcoremock v0.1.3
```

After replacing "hashicorp/tfcoremock" with "tfcoremock" in main.tf:
```
~/go/bin/tofu -version
OpenTofu v1.6.0-dev
on linux_amd64
+ provider registry.opentofu.org/hashicorp/tfcoremock v0.1.3
+ provider registry.opentofu.org/opentofu/tfcoremock v0.2.0
```

**Comment by sdlarsen** [comment]
> That's fascinating. On initial inspection I'm not seeing any changes that seem like they would be directly related: [v1.6.0-beta2...v1.6.0-beta3](https://github.com/opentofu/opentofu/compare/v1.6.0-beta2...v1.6.0-beta3)
> 
> However I am seeing: `+ provider registry.opentofu.org/hashicorp/aws v5.24.0` and `+ provider registry.opentofu.org/opentofu/aws v5.30.0`
> 
> That looks like a good place to start, and could be related to changes in how we handle the default namespace.
> 
> Could you check on beta2 to see what providers and versions are listed?

Not sure this is still needed, but here it is for beta2:
```OpenTofu v1.6.0-beta2
on linux_amd64
+ provider registry.opentofu.org/community-terraform-providers/ignition v2.2.2
+ provider registry.opentofu.org/datadog/datadog v3.33.0
+ provider registry.opentofu.org/gitlabhq/gitlab v16.6.0
+ provider registry.opentofu.org/hashicorp/archive v2.4.0
+ provider registry.opentofu.org/hashicorp/aws v5.24.0
+ provider registry.opentofu.org/hashicorp/local v2.4.0
+ provider registry.opentofu.org/hashicorp/null v3.2.1
+ provider registry.opentofu.org/hashicorp/template v2.2.0
+ provider registry.opentofu.org/hashicorp/tls v4.0.4
+ provider registry.opentofu.org/opentofu/aws v5.30.0
+ provider registry.opentofu.org/opentofu/null v3.2.2
```

**Comment by cam72cam** [comment]
Thanks, that confirms my initial investigation was on the right track and that my PR would address it.  I'll be meeting with some other core engineers tomorrow to dive deeper into exactly how we want to address this problem.  Thanks for the report!
